pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -119,12 +165,21 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 			if port.Protocol == protocol.UDP {
 				continue
 			}
-			lbEndpoints := cb.buildLocalityLbEndpoints(networkView, service, port.Port, nil)
+			clusterKey := buildClusterKey(service, port, cb)
+			cached, tokens, allFound := cb.getAllCachedSubsetClusters(*clusterKey)
+			if allFound && !features.EnableUnsafeAssertions {
+				hit += len(cached)
+				resources = append(resources, cached...)
+				continue
+			} else {
```

Comment:
I don't really understand why we increment by the length of all the clusters

shouldn't we only increment `hit` or `miss` by the number of clusters that were a "hit" or "miss"? why does allfound matter?

Reply:
the length of all the clusters is == the number of hit or miss. They either all hit or they all miss because we require all subsets to be a hit for it to be considered a hit